internal/pkg/db/mongo/device_profile.go

```diff
@@ -101,16 +102,22 @@ func (mdp *mongoDeviceProfile) SetBSON(raw bson.Raw) error {
 
 	comCol := s.DB(m.database.Name).C(db.Command)
 
-	var commands []models.Command
+	var commands []contract.Command
 
 	// Get all of the commands from the references
 	for _, cRef := range decoded.Commands {
 		var c models.Command
-		err := comCol.FindId(cRef.Id).One(&c)
+
+		query := comCol.FindId(cRef.Id)
+		if n, _ := query.Count(); n > 0 {
```

Comment:
Bear in mind we want to eventually move this database access code out of this model (see issue #878). It's fine to make these short term edits in order to satisfy blackbox tests and ensure we have clean functionality.

Reply:
Moved database code out of mongoDeviceProfile model.